Mathlib/RingTheory/DividedPowers/DPMorphism.lean

```diff
@@ -0,0 +1,242 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández
+-/
+import Mathlib.RingTheory.DividedPowers.Basic
+
+/-! # Divided power morphisms
+
+Let `A` and `B` be commutative (semi)rings, let `I` be an ideal of `A` and let `J` be an ideal of
+`B`. Given divided power structures on `I` and `J`, a ring morphism `A →+* B` is a *divided
+power morphism* if it is compatible with these divided power structures.
+
+## Main definitions
+
+* `DividedPowers.IsDPMorphism` : given divided power structures on the `A`-ideal `I` and the
+  `B`-ideal `J`, a ring morphism `A →+* B` is a divided power morphism if it is compatible with
+  these divided power structures.
+* `DividedPowers.DPMorphism` : a bundled version of `IsDPMorphism`.
+* `DividedPowers.ideal_from_ringHom` : given a ring homomorphism `A →+* B` and ideals `I ⊆ A` and
+  `J ⊆ B` such that `I.map f ≤ J`, this is the `A`-ideal on which
+  `f (hI.dpow n x) = hJ.dpow n (f x)`.
+* `DividedPowers.DPMorphism.fromGens` : the `DPMorphism` induced by a ring morphism, given that
+  divided powers are compatible on a generating set.
+
+## Main results
+
+* `DividedPowers.dpow_eq_from_gens` : if two divided power structures on an ideal `I` agree on a
+  generating set, then they are equal.
+
+
+## References
+
+* [P. Berthelot (1974), *Cohomologie cristalline des schémas de
```

Comment:
You can add those to the references.bib file.

Reply:
These are already in the .bib file. I will add links to the specific results below.